imports/histogram/reducer.js

```diff
@@ -1,15 +1,16 @@
+// @flow
 // import { combineReducers } from 'redux';
-import { ActionType } from './actions';
+import type { Action } from '../shared/action.models';
 
 // const defaultData = {
 //   // rootDir: '/tmp',
 //   // files: [{ name: 'apple.fits', type: 'fits' }],
 // };
 const defaultState = { data: {} };
 
-const HistogramDB = (state = defaultState, action) => {
+const HistogramDB = (state: {data: any} = defaultState, action: Action) => {
   switch (action.type) {
-    case ActionType.HISTOGRAM_CHANGE: {
+    case 'HISTOGRAM_CHANGE': {
```

Comment:
Here you follow https://flow.org/en/docs/react/redux/ example which needs to type an action-type twice. A normal pattern used in redux is, you define an action type somewhere, and it can be used in action creator, also in reducer. example, https://redux.js.org/docs/basics/Reducers.html. Using a const variable is better than typing the same string twice.